src/main/webapp/WEB-INF/pages/product.jsp

```diff
@@ -5,6 +5,21 @@
 
 <jsp:useBean id="product" type="com.es.phoneshop.model.product.Product" scope="request"/>
 <tags:master pageTitle="Product Details">
+
+    <p>
+        cart: ${cart}
+    </p>
+    <c:if test="${not empty param.message}">
```

Comment:
Correct situation when after successful add to cart user make error. In such case both success and error messages are displayed.